docs/madr/decisions/034-default-target-ref-policy.md

```diff
@@ -0,0 +1,135 @@
+# TargetRef policies as a default
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/8467
+
+## Context and Problem Statement
+
+At present, Kuma has two policy types:
+
+* Legacy policies with Source and Destination attributes
+* New policies based on TargetRef
+ 
+The legacy policies currently serve as the defaults. However, starting from release 2.6, the intention is to switch the default policy to TargetRef.
+
+## Considered Options
+
+1. Prefer avoiding the creation of default policies and utilize plugin code in cases where there are no existing default legacy policy.
+2. Introduce a `policyEngine` field within the `Mesh` object, allowing the selection of the default policy version.
+
+## Decision Outcome
+
+The decision is to stop creating default policies and utilize plugin code in cases where there are no existing default legacy policy.
+
+### Positive Consequences
+
+* New deployments will start with the new policy engine.
+* Existing meshes remain unaffected by the change.
+* No additional configuration.
+
+### Negative Consequences
+
+* Clients who are recreating their deployment using continuous delivery (CD) need to manually add default `Timeout`, `CircuitBreaker`, and `Retry` policies
+
+## Pros and Cons of the Options
+
+### Prefer avoiding the creation of default policies and utilize plugin code in cases where there are no existing default legacy policy.
+
+As of the 2.6 release, we have discontinued the creation of default policies when initializing a new mesh. This change in the plugin's code allows us to verify the existence of an older matching policy. In cases where there is no existing policy, we generate the configuration using the plugin code.
+
+To support older versions of zones we are going to introduce `KUMA_DEFAULTS_CREATE_MESH_RESOURCES` which by default is disabled. The user can enable `KUMA_DEFAULTS_CREATE_MESH_RESOURCES` and that allows new meshes to be created with default policies.
```

Comment:
so what's the case for this? someone upgrades global to 2.6 and the zones are on 2.5, the global no longer creates default policies (but I guess it does not remove the existing ones, right?) so nothing changes? then someone updates the zones to 2.6 and still nothing changes 🤔 or are you thinking about someone creating a new mesh while global is on 2.6 and zone is on 2.5 and the new mesh does not work because there are no default policies? 🤔 should we care about that case?

Reply:
> are you thinking about someone creating a new mesh while global is on 2.6 and zone is on 2.5 and the new mesh does not work because there are no default policies
It's about this one and I think we should care about it.

Reply:
> a new mesh while global is on 2.6 and zone is on 2.5 and the new mesh does not work because there are no default policies

yeah, you're right because we have hosted global and can't control zone upgrades

Reply:
Added note

Reply:
I'm a bit sceptical whether we should introduce this env. This can confuse users further what to do.

What are the chances that in the middle of Kuma upgrade someone creates a Mesh, spins up new services and DPPs in the mesh and cannot afford this behavior?

Reply:
I agree with @jakubdyszkiewicz that introducing a separate environment variable might introduce unnecessary complexity and confusion, potentially outweighing its marginal benefits. Instead, we should focus on providing clear and concise documentation that guides users through this specific behavior.

Reply:
What about multitenant deployment when clients might not be at the same versions? In this case, it's easier to allow the creation of old ones policies to make everything works